src/bors/handlers/review.rs

```diff
@@ -598,4 +646,220 @@ approve = ["+approved"]
         })
         .await;
     }
+
+    fn reviewer() -> User {
+        User::new(10, "reviewer")
+    }
+
+    fn as_reviewer(text: &str) -> Comment {
+        Comment::from(text).with_author(reviewer())
+    }
+
+    fn create_world_with_delegate_config() -> World {
+        let world = World::default();
+        world.default_repo().lock().set_config(
+            r#"
+[labels]
+approve = ["+approved"]
+"#,
+        );
+        world.default_repo().lock().permissions = Permissions::default();
+        world
+            .default_repo()
+            .lock()
+            .permissions
+            .users
+            .insert(reviewer(), vec![PermissionType::Review]);
+
+        world
+    }
+
+    #[sqlx::test]
+    async fn delegate_author(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .world(create_world_with_delegate_config())
+            .run_test(|mut tester| async {
+                tester.post_comment(as_reviewer("@bors delegate+")).await?;
+                assert_eq!(
+                    tester.get_comment().await?,
+                    format!(
+                        "@{} can now approve this pull request",
+                        User::default().name
+                    )
+                );
+
+                let pr = tester.get_default_pr().await?;
+                assert!(pr.delegated);
+                Ok(tester)
+            })
+            .await;
+    }
+
+    #[sqlx::test]
+    async fn delegatee_can_approve(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .world(create_world_with_delegate_config())
+            .run_test(|mut tester| async {
+                tester.post_comment(as_reviewer("@bors delegate+")).await?;
+                tester.expect_comments(1).await;
+
+                tester.post_comment("@bors r+").await?;
+                tester.expect_comments(1).await;
+
+                check_pr_approved_by(&tester, default_pr_number().into(), &User::default().name)
+                    .await;
+                Ok(tester)
+            })
+            .await;
+    }
+
+    #[sqlx::test]
+    async fn delegatee_can_try(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .world(create_world_with_delegate_config())
+            .run_test(|mut tester| async {
+                tester.post_comment(as_reviewer("@bors delegate+")).await?;
+                tester.expect_comments(1).await;
+
+                tester.post_comment("@bors try").await?;
+                tester.expect_comments(1).await;
+                tester
```

Comment:
This actually sends an event to bors that the workflow has started (it doesn't check that it has started). Check e.g. `try_merge_branch_history` for how can we check that a try build has been requested.